src/libstd/sys/unix/fd.rs

```diff
@@ -9,6 +9,11 @@ use crate::sys_common::AsInner;
 use libc::{c_int, c_void};
 
 #[derive(Debug)]
+#[rustc_layout_scalar_valid_range_start(0)]
+// libstd/os/raw/mod.rs assures me that every Rust-supported platform has a
+// 32-bit c_int. Below is -2, in two's complement, but that only works out
+// because c_int is 32 bits.
+#[rustc_layout_scalar_valid_range_end(4294967294)]
```

Comment:
Is it possible to express this in hex? `0xFFFFFFFE` would be much more readable.